The Python 2.7 build fails on Travis

A few tests fail with an error like this one:
```
==================================== ERRORS ====================================
_________________ ERROR collecting tests/test_active_cells.py __________________
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/python.py:507: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/assertion/rewrite.py:304: in load_module
    exec(co, mod.__dict__)
tests/test_active_cells.py:2: in <module>
    from jupytext.compare import compare
jupytext/__init__.py:3: in <module>
    from .jupytext import read, write, readf, writef, writes, reads
jupytext/jupytext.py:12: in <module>
    from .formats import _VALID_FORMAT_OPTIONS
jupytext/formats.py:10: in <module>
    from .cell_reader import MarkdownCellReader, RMarkdownCellReader, \
jupytext/cell_reader.py:9: in <module>
    from sphinx_gallery.notebook import rst2md
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/sphinx_gallery/notebook.py:20: in <module>
    from .py_source_parser import split_code_and_text_blocks
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/sphinx_gallery/py_source_parser.py:18: in <module>
    from .sphinx_compatibility import getLogger
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/sphinx_gallery/sphinx_compatibility.py:12: in <module>
    import sphinx.util
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/sphinx/util/__init__.py:29: in <module>
    from docutils.utils import relative_path
../../../virtualenv/python2.7.14/lib/python2.7/site-packages/docutils/utils/__init__.py:21: in <module>
    import docutils.io
E     File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/docutils/io.py", line 348
E       (self.destination.mode, mode)), file=self._stderr)
E                                           ^
E   SyntaxError: invalid syntax
```

--- a/jupytext/cell_reader.py
+++ b/jupytext/cell_reader.py
@@ -5,9 +5,10 @@ import json
 from nbformat.v4.nbbase import new_code_cell, new_raw_cell, new_markdown_cell
 from .languages import _SCRIPT_EXTENSIONS
 
+# Sphinx Gallery is an optional dependency. And we intercept the SyntaxError for #301
 try:
     from sphinx_gallery.notebook import rst2md
-except ImportError:
+except (ImportError, SyntaxError):
     rst2md = None
 
 from .cell_metadata import is_active, json_options_to_metadata, md_options_to_metadata, rmd_options_to_metadata, \

